api/envoy/config/filter/http/fault/v2/fault.proto

```diff
@@ -21,13 +21,22 @@ option (udpa.annotations.file_migrate).move_to_package = "envoy.extensions.filte
 // [#extension: envoy.filters.http.fault]
 
 message FaultAbort {
+  // Fault aborts are controlled via an HTTP header (if applicable). See the
+  // :ref:`HTTP fault filter <config_http_filters_fault_injection_http_header>` documentation for
+  // more information.
+  message HeaderAbort {
+  }
+
   reserved 1;
 
   oneof error_type {
     option (validate.required) = true;
 
     // HTTP status code to use to abort the HTTP request.
     uint32 http_status = 2 [(validate.rules).uint32 = {lt: 600 gte: 200}];
+
+    // Fault aborts are controlled via an HTTP header (if applicable).
+    HeaderAbort header_abort = 4;
```

Comment:
I would call this `header_error`. The abort is still controlled by the normal mechanisms, it's just the nature of the error is expressed in the header.

Reply:
I can change it but I think that `header_abort` is a better name because:
1. `header_error` can be in theory used to inject status codes such as `200` or `204` that are not errors.
1. `header_abort` follows a naming scheme used by [`header_limit`](https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/fault/v2/fault.proto#L80) and [`header_delay`](https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/fault/v2/fault.proto#L52) fields. `header_error` would've been a better choice if `header_limit` was `header_kbs` and `header_delay` was `header_duration`.

Thoughts?

Reply:
I think I would be inclined to stick with `header_abort` for the reasons that @Augustyniak outlines. @htuch WDYT?

Reply:
Fair enough.